javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -498,13 +498,28 @@ public CharSeq padTo(int length, Character element) {
             return this;
         }
         final StringBuilder sb = new StringBuilder(back);
-        final int limit = length - back.length();
-        for (int i = 0; i < limit; i++) {
-            sb.append(element);
+        sb.append(padding(element, length - back.length()));
+        return new CharSeq(sb.toString());
+    }
+
+    @Override
+    public CharSeq leftPadTo(int length, Character element) {
+        if (length <= back.length()) {
+            return this;
         }
+        final StringBuilder sb = new StringBuilder(padding(element, length - back.length()));
+        sb.append(back);
         return new CharSeq(sb.toString());
     }
 
+    private static StringBuilder padding(Character element, int limit) {
```

Comment:
let's change signature to 
```java
private static void padding(StringBuilder sb, Character element, int limit)
```
to avoid sreating new StringBuilder instance